src/timing.h

```diff
@@ -406,16 +361,23 @@ STATIC_INLINE void _jl_timing_suspend_destroy(jl_timing_suspend_t *suspend) JL_N
 #endif
 }
 
-#define JL_TIMING(subsystem, event) \
-    JL_TIMING_CREATE_BLOCK(__timing_block, subsystem, event); \
+#define JL_TIMING(subsystem, event)                                                \
+    JL_TIMING_CREATE_BLOCK(__timing_block, subsystem, event);                      \
     jl_timing_block_start(&__timing_block)
 
-#define JL_TIMING_CREATE_BLOCK(block, subsystem, event) \
-    __attribute__((cleanup(_jl_timing_block_destroy))) \
-    jl_timing_block_t block; \
-    _jl_timing_block_ctor(&block, JL_TIMING_ ## subsystem, JL_TIMING_EVENT_ ## event); \
-    _TRACY_CTOR(&block, #event); \
-    _ITTAPI_CTOR(&block, #event)
+#define JL_TIMING_CREATE_BLOCK(block, subsystem_name, event_name)                  \
+    static jl_timing_event_t TIMING_CONCAT(__timing_event, __LINE__) = { 0 };      \
```

Comment:
I would rather this be routed through jl_timing_event_create, so we only have one implementation to worry about.

Reply:
That's a really good idea, let me do that